src/aspire/basis/pswf_2d.py

```diff
@@ -1,5 +1,320 @@
+
+        while True:
+            alpha = self.alpha_all[m]
+
+            lambda_var = a * np.square(np.absolute(alpha))
+            gamma = np.sqrt(np.absolute(lambda_var / (1 - lambda_var)))
+
+            n_end = np.where(gamma <= self.truncation)[0]
+
+            if len(n_end) != 0:
+                n_end = n_end[0]
+                if n_end == 0:
+                    break
+                max_ns.extend([n_end])
+                alpha_all.extend(alpha[:n_end])
+                m += 1
+
+        if self.even:
+            x_1d_grid = range(-self.resolution, self.resolution)
+        else:
+            x_1d_grid = range(-self.resolution, self.resolution + 1)
+            
+        x_2d_grid, y_2d_grid = np.meshgrid(x_1d_grid, x_1d_grid)
+        r_2d_grid = np.sqrt(np.square(x_2d_grid) + np.square(y_2d_grid))
+        points_inside_the_circle = r_2d_grid <= self.resolution
+        x = y_2d_grid[points_inside_the_circle]
+        y = x_2d_grid[points_inside_the_circle]
+
+        r_2d_grid_on_the_circle = np.sqrt(np.square(x) + np.square(y)) / self.resolution
+        theta_2d_grid_on_the_circle = np.angle(x + 1j * y)
+
+        self.samples = self.evaluate_pswf2d_all(r_2d_grid_on_the_circle, theta_2d_grid_on_the_circle, max_ns)
+
+        self.angular_frequency = np.repeat(np.arange(len(max_ns)), max_ns).astype('float')
+        self.radian_frequency = np.concatenate([range(1, l + 1) for l in max_ns]).astype('float')
+        self.alpha_nn = np.array(alpha_all)
+        self.samples = (self.beta / 2.0) * self.samples * self.alpha_nn
+        #self.samples = (self.beta / 2.0) * self.samples * self.alpha_nn.conj()
+        self.samples_conj_transpose = self.samples.conj().transpose()
+
+        self.image_height = len(x_1d_grid)
+
+        self.points_inside_the_circle = points_inside_the_circle
+
+        self.points_inside_the_circle_vec = points_inside_the_circle.reshape(self.image_height ** 2)
+
+        self.non_neg_freq_inds = slice(0, len(self.angular_frequency))
+
+        tmp = np.nonzero(self.angular_frequency == 0)[0]
+        self.zero_freq_inds = slice(tmp[0], tmp[-1] + 1)
+
+        tmp = np.nonzero(self.angular_frequency > 0)[0]
+        self.pos_freq_inds = slice(tmp[0], tmp[-1] + 1)
+
+    def get_points_inside_the_circle(self):
+        return self.points_inside_the_circle
+
+    def mask_points_inside_the_circle(self, images):
+        return self.__mask_points_inside_the_circle_cpu(images)
+
+    def __mask_points_inside_the_circle_cpu(self, images):
+        return images * self.points_inside_the_circle
+
+    def evaluate_t(self, images):
+        images_shape = images.shape
+
+        # if we got several images
+        if len(images_shape) == 3:
+            flattened_images = images.reshape((images_shape[0] * images_shape[1], images_shape[2]), order='F')
+
+        # else we got only one image
+        else:
+            flattened_images = images.reshape((images_shape[0] * images_shape[1], 1), order='F')
```

Comment:
Remove this line and add `images_shape = (images_shape + (1,)) if len(images_shape) == 2 else images_shape` before.

Reply:
done